### DISCUSSION TITLE
Would Grafana be interested in supporting react-native?

### DISCUSSION THREAD

**Discussion by edkimmel** [grafana/faro-web-sdk#706] [Ideas]
I did some work recently to integrate this sdk into our react native (and react-native-web) application.  It required a few minor patches to both core and web.  95% of the web sdk code worked out of the box with react native, but there are a few differences.  
Would Grafana be interested in maintaining this support upstream?

- BatchExecutor uses document and arguably should be an implementation provided by the web sdk.  Allowing a custom batch executor implementation would let a `faro-react-native-sdk` override it with equivalent RN apis
- `faro-react-native-sdk` would not include the browser / page meta by default, but could still have optional react-navigation integration to help
- `faro-react-native-sdk` would use one of the RN storage libraries out there instead for session management, or expect an implementation to be provided in the config to allow the app to stay in control on which library it wants to bundle with.

**Comment by codecapitano** [comment]
Hi @edkimmel thank you so much and yes we would like to support RN some day but didn't tackle it yet because of conflicting priorities.

However our team member @eskirk already started a spike on it and I would like to connect you here for further discussion.

Cheers,
Marco

**Comment by eskirk** [comment]
hi @edkimmel just to shed some more light on this- 

I have an outdated, [very rough draft PR](https://github.com/grafana/faro-web-sdk/pull/241) of a react native instrumentation, but have not been able to dedicate any time to it since my original attempt. 

there are some talks at Grafana about potentially reviving this, but we do not have any official plans as of now. I would love to be able to revive this a bit, but not sure when that will be. 

if you were interested in sharing the work that you have done I imagine there could be a path forward where we may be able to use that? let us know and thanks for keeping this on our radar.
